resources/prosody-plugins/mod_xxl_conference.lua

```diff
@@ -0,0 +1,222 @@
+--- activate under main vhost
+--- In /etc/hosts add:
+--- vm1-ip-address visitors1.domain.com
+--- vm1-ip-address conference.visitors1.domain.com
+--- vm2-ip-address visitors2.domain.com
+--- vm2-ip-address conference.visitors2.domain.com
+--- TODO: drop the /etc/hosts changes for https://modules.prosody.im/mod_s2soutinjection.html
+--- Enable in global modules: 's2s_bidi' and 'certs_all'
+--- Make sure 's2s' is not in modules_disabled
+--- Open port 5269 on the provider side and on the firewall on the machine (iptables -I INPUT 4 -p tcp -m tcp --dport 5269 -j ACCEPT)
+--- TODO: make it work with tenants
+local st = require 'util.stanza';
+local jid = require 'util.jid';
+local util = module:require 'util';
+local presence_check_status = util.presence_check_status;
+
+local um_is_admin = require 'core.usermanager'.is_admin;
+local function is_admin(jid)
+    return um_is_admin(jid, module.host);
+end
+
+local MUC_NS = 'http://jabber.org/protocol/muc';
+
+-- get/infer focus component hostname so we can intercept IQ bound for it
+local focus_component_host = module:get_option_string('focus_component');
+if not focus_component_host then
+    local muc_domain_base = module:get_option_string('muc_mapper_domain_base');
+    if not muc_domain_base then
+        module:log('error', 'Could not infer focus domain. Disabling %s', module:get_name());
+        return;
+    end
+    focus_component_host = 'focus.'..muc_domain_base;
+end
+
+-- required parameter for custom muc component prefix, defaults to 'conference'
+local muc_domain_prefix = module:get_option_string('muc_mapper_domain_prefix', 'conference');
+
+local main_muc_component_config = module:get_option_string('main_muc');
+if main_muc_component_config == nil then
+    module:log('error', 'xxl rooms not enabled missing main_muc config');
+    return ;
+end
+
+function table_length(T)
+    local count = 0
+    for _ in pairs(T) do count = count + 1 end
```

Comment:
At least you didn't need to implement integer addition

Reply:
And this seems unused ...